test/components/plots/areaPlotTests.ts

```diff
@@ -113,17 +113,54 @@ describe("Plots", () => {
       svg.remove();
     });
 
-    it("getAllSelections retrieves correct selections", () => {
-      var newTwoPointData = [{foo: 2, bar: 1}, {foo: 3, bar: 2}];
-      var newDataset = new Plottable.Dataset(twoPointData);
-      areaPlot.addDataset(new Plottable.Dataset(newTwoPointData));
-      var allAreas = areaPlot.getAllSelections();
-      assert.strictEqual(allAreas.size(), 2, "all areas retrieved");
-      var selectionData = allAreas.data();
-      assert.include(selectionData, twoPointData, "first dataset data in selection data");
-      assert.include(selectionData, newTwoPointData, "new dataset data in selection data");
-
-      svg.remove();
+    describe("getAllSelections()", () => {
+
+      it("retrieves all selections with no args", () => {
+        var newTwoPointData = [{ foo: 2, bar: 1 }, { foo: 3, bar: 2 }];
+        var newDataset = new Plottable.Dataset(twoPointData);
+        areaPlot.addDataset("newTwo", new Plottable.Dataset(newTwoPointData));
+        var allAreas = areaPlot.getAllSelections();
+        var allAreas2 = areaPlot.getAllSelections((<any> areaPlot)._datasetKeysInOrder);
```

Comment:
I specifically wanted to assert that it returned the same thing that getAllSelections with the same all dataset keys did, rather than assert the ground truth.  Do you think it should instead assert the ground truth?

The way it is now ties the behavior of the two functions whereas changing it will essentially test each one independently.  It is mainly a  matter of which behavior we want ourselves.